mithril-stm/src/multi_sig.rs

```diff
@@ -200,10 +200,11 @@ impl From<&SigningKey> for ProofOfPossession {
         use blst::blst_sk_to_pk_in_g1;
         let k1 = sk.0.sign(POP, &[], &[]);
         let k2 = unsafe {
-            let sk_scalar = std::mem::transmute::<&BlstSk, &blst_scalar>(&sk.0);
-
+            let ser_sk = sk.0.serialize().as_ptr();
+            let mut sk_scalar = blst_scalar::default();
+            blst_scalar_from_bendian(&mut sk_scalar, ser_sk);
             let mut out = blst_p1::default();
-            blst_sk_to_pk_in_g1(&mut out, sk_scalar);
+            blst_sk_to_pk_in_g1(&mut out, &sk_scalar);
```

Comment:
In general I would avoid doing serialisation/deserialisation, and use it as a last resort. So if for the other usages of transmute we can use different strategies, it'll be best. 

If we were to continue with this technique (of serialisation and deserialisation), I would be careful with secret material. i.e. I would overwrite `ser_sk` and `sk_scalar` with zeroes and `blst_scalar::default` respectively.